src/mlpack/methods/neighbor_search/neighbor_search_rules_impl.hpp

```diff
@@ -375,6 +375,9 @@ inline double NeighborSearchRules<SortPolicy, MetricType, TreeType>::Rescore(
   if (oldScore == DBL_MAX)
     return oldScore;
 
+  if (oldScore == SortPolicy::BestDistance())
+    return oldScore;
```

Comment:
@rcurtin 
Yes, I completely agree. I added this modification to avoid calculating the bounds as there is no possibility of prunning when score is `0`. I agree that it would be possible to prune when score is 0, but we can't because of `B_2()`.
Updated the code in : https://github.com/mlpack/mlpack/pull/747/commits/e4cf1fd252b710809654515dccc30a299d0380e2